tests/components/homekit_controller/common.py

```diff
@@ -0,0 +1,140 @@
+        self.accessory = accessory
+        self.pairing_data = {
+            'accessories': self.list_accessories_and_characteristics()
+        }
+
+    def list_accessories_and_characteristics(self):
+        """Fake implementation of list_accessories_and_characteristics."""
+        return [self.accessory.to_accessory_and_service_list()]
+
+    def get_characteristics(self, characteristics):
+        """Fake implementation of get_characteristics."""
+        results = {}
+        for aid, cid in characteristics:
+            for service in self.accessory.services:
+                for char in service.characteristics:
+                    if char.iid != cid:
+                        continue
+                    results[(aid, cid)] = {
+                        'value': char.get_value()
+                    }
+        return results
+
+    def put_characteristics(self, characteristics):
+        """Fake implementation of put_characteristics."""
+        for _, cid, new_val in characteristics:
+            for service in self.accessory.services:
+                for char in service.characteristics:
+                    if char.iid != cid:
+                        continue
+                    char.set_value(new_val)
+
+
+class FakeController:
+    """
+    A test fake that pretends to be a paired HomeKit accessory.
+
+    This only contains methods and values that exist on the upstream Controller
+    class.
+    """
+
+    def __init__(self):
+        """Create a Fake controller with no pairings."""
+        self.pairings = {}
+
+    def add(self, accessory):
+        """Create and register a fake pairing for a simulated accessory."""
+        pairing = FakePairing(accessory)
+        self.pairings['00:00:00:00:00:00'] = pairing
+        return pairing
+
+
+class Helper:
+    """Helper methods for interacting with HomeKit fakes."""
+
+    def __init__(self, hass, entity_id, pairing, accessory):
+        """Create a helper for a given accessory/entity."""
+        from homekit.model.services import ServicesTypes
+        from homekit.model.characteristics import CharacteristicsTypes
+
+        self.hass = hass
+        self.entity_id = entity_id
+        self.pairing = pairing
+        self.accessory = accessory
+
+        self.characteristics = {}
+        for service in self.accessory.services:
+            service_name = ServicesTypes.get_short(service.type)
+            for char in service.characteristics:
+                char_name = CharacteristicsTypes.get_short(char.type)
+                self.characteristics[(service_name, char_name)] = char
+
+    async def get_state(self):
+        """Trigger an entity update() and return the current entity state."""
+        await self.hass.helpers.entity_component.async_update_entity(
+            self.entity_id)
+        state = self.hass.states.get(self.entity_id)
+        assert state is not None
+        return state
+
+
+async def add_fake_homekit_accessory(hass, services):
```

Comment:
It's not clear from the name that this will also set up the component. Maybe break that out, or rename this function?